### ISSUE TITLE
Calculate image parameters in telescope frame and use effective focal length

### ISSUE THREAD

**Issue by maxnoe** [cta-observatory/cta-lstchain#988]
This will properly calculate hillas and other parameters in degrees and if used together with the effective focal length applies the coma correction.

This should improve the off-axis performance a lot.

**Comment by maxnoe** [comment]
Related: https://github.com/cta-observatory/ctapipe/pull/1917

**Comment by moralejo** [comment]
> This will properly calculate hillas and other parameters in degrees and if used together with the effective focal length applies the coma correction.
> 

ok!

> This should improve the off-axis performance a lot.

I do not think it will be the case for mono analysis. In stereo the main problem is you get the geometrical reconstruction wrong, i.e. get wrong impact parameters & Hmax. That then enters all the other reconstructions.

In mono we will get a shift in the reconstructed direction but as long as we are doing the same in data and MC (and we are, i.e. using the nominal focal length for the transformations) that will be the only bad effect. I don't think RF performance (for energy reco & g/h separation) will be affected. It is like using different units for all camera quantities.

The shift in reco direction can be taken into account at higher level in the analysis and is anyway pretty small compared to angular resolution for the standard wobble offset (0.4 deg, hence 0.4*1.0466 = 0.4186,  or  0.0186 deg shift)

**Comment by maxnoe** [comment]
@moralejo Yes, agreed for mono, but the dl1 files are also used by the combined magic + lst analysis, right?

**Comment by moralejo** [comment]
Yes, indeed! Also, in mono the gamma spot offset will be more significant if you go e.g. 1 deg off-axis or more. I was thinking on the effect in our standard observations, and that one will be small.

**Comment by vuillaut** [comment]
Hi
I wanted to open a new issue, but it's actually a revival of this one.
The `equivalent focal length` is used at R0 to DL1 stage:

https://github.com/cta-observatory/cta-lstchain/blob/67c1c32906dc3242554987e03c3220654388b72c/lstchain/reco/r0_to_dl1.py#L259

https://github.com/cta-observatory/cta-lstchain/blob/67c1c32906dc3242554987e03c3220654388b72c/lstchain/reco/r0_to_dl1.py#L651

---

But then disp parameters are recomputed at DL1 to DL2 stage using the effective focal length:

https://github.com/cta-observatory/cta-lstchain/blob/67c1c32906dc3242554987e03c3220654388b72c/lstchain/reco/dl1_to_dl2.py#L527

---

Shouldn't the effective focal length be used everywhere, so we can avoid this recomputation?

Note: maybe the re-computation was introduced to avoid recomputing all the DL1 parameters?
